components/director/internal/formationmapping/auth_middleware.go

```diff
@@ -276,6 +278,9 @@ func (a *Authenticator) isFormationAuthorized(ctx context.Context, formationID s
 
 	f, err := a.formationRepo.GetGlobalByID(ctx, formationID)
 	if err != nil {
+		if apperrors.IsNotFoundError(err) {
+			return false, http.StatusBadRequest, errors.Wrapf(err, "while getting formation with ID: %q globally", formationID)
+		}
```

Comment:
Isn't the 404 status more appropriate in such cases instead of a bad request? Also, the error message should be adapted as well because currently all of them are `errors.Errorf("An unexpected error occurred while processing the request. X-Request-Id: %s", correlationID)` regardless of the status code
P.S. valid for other places as well